mos-platform/atari8-mega/link.ld

```diff
@@ -0,0 +1,193 @@
+
+MEMORY {
+  /* bank0 is truncated by 20 bytes so we can add a tail */
+  bank0         : ORIGIN = __bank0_lma,  LENGTH = 0x4000 - 20
+  tail0        : ORIGIN = __bank0_lma + 0x4000 - 20, LENGTH = 20
+
+  bank1         : ORIGIN = __bank1_lma,  LENGTH = __cart_rom_size >=  32 ? 0x4000 : 0
+  bank2         : ORIGIN = __bank2_lma,  LENGTH = __cart_rom_size >=  64 ? 0x4000 : 0
+  bank3         : ORIGIN = __bank3_lma,  LENGTH = __cart_rom_size >=  64 ? 0x4000 : 0
+  bank4         : ORIGIN = __bank4_lma,  LENGTH = __cart_rom_size >= 128 ? 0x4000 : 0
+  bank5         : ORIGIN = __bank5_lma,  LENGTH = __cart_rom_size >= 128 ? 0x4000 : 0
+  bank6         : ORIGIN = __bank6_lma,  LENGTH = __cart_rom_size >= 128 ? 0x4000 : 0
+  bank7         : ORIGIN = __bank7_lma,  LENGTH = __cart_rom_size >= 128 ? 0x4000 : 0
+
+  bank8         : ORIGIN = __bank8_lma,  LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank9         : ORIGIN = __bank9_lma,  LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank10        : ORIGIN = __bank10_lma, LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank11        : ORIGIN = __bank11_lma, LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank12        : ORIGIN = __bank12_lma, LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank13        : ORIGIN = __bank13_lma, LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank14        : ORIGIN = __bank14_lma, LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+  bank15        : ORIGIN = __bank15_lma, LENGTH = __cart_rom_size >= 256 ? 0x4000 : 0
+
+  bank16        : ORIGIN = __bank16_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank17        : ORIGIN = __bank17_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank18        : ORIGIN = __bank18_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank19        : ORIGIN = __bank19_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank20        : ORIGIN = __bank20_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank21        : ORIGIN = __bank21_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank22        : ORIGIN = __bank22_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank23        : ORIGIN = __bank23_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank24        : ORIGIN = __bank24_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank25        : ORIGIN = __bank25_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank26        : ORIGIN = __bank26_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank27        : ORIGIN = __bank27_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank28        : ORIGIN = __bank28_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank29        : ORIGIN = __bank29_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank30        : ORIGIN = __bank30_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank31        : ORIGIN = __bank31_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+}
+
+REGION_ALIAS("c_writeable", ram)
+REGION_ALIAS("c_readonly", bank0)
```

Comment:
We've kept as an invariant that the default C sections are always placed in a location that is always visible, regardless of the banking state. It's generally not possible to write C safely without this guarantee, since the compiler assumes that library functions are always callable, there's no real specification for when a compiler will emit a library call, and the library call implementations are placed in the default C sections.

We did come up with a solution in the `nes-gtrom` linker scripts which is a broadly similar situation; the trick is to duplicate the contents of the fixed bank into each variable one. This wastes space, but it allows plain C to work as expected, while still making it possible to move functions and rodata out of the duplicated fixed region into specific banks. We'll need to do the same here.

Reply:
Will this not end up in all banks though?  I wanted to be able to pick-and-choose which banks had code vs data.

Reply:
Perhaps it would be possible to have the first `m` banks be code with the fixed region, and the remaining `n-m` banks be for data only.

Reply:
@mysterymath , @asiekierka PTAL